x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ForceMergeAction.java

```diff
@@ -112,7 +169,8 @@ public boolean equals(Object obj) {
             return false;
         }
         ForceMergeAction other = (ForceMergeAction) obj;
-        return Objects.equals(maxNumSegments, other.maxNumSegments);
+        return Objects.equals(maxNumSegments, other.maxNumSegments)
+            && ((codec == null && other.codec == null) || Objects.equals(codec, other.codec));
```

Comment:
No need for the null check

```suggestion
            && Objects.equals(codec, other.codec);
```